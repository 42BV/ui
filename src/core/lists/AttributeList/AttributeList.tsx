import React from 'react';
import { ListGroup } from 'reactstrap';

type Props = {
  /**
   * The content of the list.
   * This should be a combination of AttributeView components.
   */
  children: React.ReactNode;
};

/**
 * AttributeList when used with AttributeView is a way to display
 * object properties in a clear and beautiful list.
 */
export function AttributeList({ children }: Props) {
  return <ListGroup flush>{children}</ListGroup>;
}
