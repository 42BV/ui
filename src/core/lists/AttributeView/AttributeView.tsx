import React from 'react';
import { ListGroupItem, ListGroupItemHeading } from 'reactstrap';

type Props = {
  /**
   * The heading to display above the value.
   */
  label: React.ReactNode;

  /**
   * The value described by the label.
   */
  children: React.ReactNode;
};

/**
 * AttributeView is used inside AttributeList to display object properties
 * in a clear and beautiful list.
 */
export function AttributeView(props: Props) {
  const { label, children } = props;

  return (
    <ListGroupItem className="pl-0">
      <ListGroupItemHeading className="small text-primary">
        {label}
      </ListGroupItemHeading>
      {children}
    </ListGroupItem>
  );
}
