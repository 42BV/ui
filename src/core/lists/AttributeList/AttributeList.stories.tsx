import { storiesOf } from '@storybook/react';

import { AttributeList } from './AttributeList';
import { AttributeView } from '../AttributeView/AttributeView';
import { Card } from 'reactstrap';

storiesOf('core/AttributeList', module)
  .addParameters({ component: AttributeList })
  .add('basic example', () => {
    return (
      <AttributeList>
        <AttributeView label="Name">42 BV</AttributeView>
        <AttributeView label="City">Zoetermeer</AttributeView>
        <AttributeView label="Country">Netherlands</AttributeView>
      </AttributeList>
    );
  })
  .add('inside card', () => {
    return (
      <Card body className="py-1">
        <AttributeList>
          <AttributeView label="Name">42 BV</AttributeView>
          <AttributeView label="City">Zoetermeer</AttributeView>
          <AttributeView label="Country">Netherlands</AttributeView>
        </AttributeList>
      </Card>
    );
  });
