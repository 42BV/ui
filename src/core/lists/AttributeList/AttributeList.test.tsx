import { render } from '@testing-library/react';

import { AttributeList } from './AttributeList';
import { AttributeView } from '../AttributeView/AttributeView';

describe('Component: AttributeList', () => {
  test('ui', () => {
    const { container } = render(
      <AttributeList>
        <AttributeView label="Name">42 BV</AttributeView>
        <AttributeView label="City">Zoetermeer</AttributeView>
        <AttributeView label="Country">Netherlands</AttributeView>
      </AttributeList>
    );

    expect(container).toMatchSnapshot();
  });
});
