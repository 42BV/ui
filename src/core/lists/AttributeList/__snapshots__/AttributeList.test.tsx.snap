// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AttributeList ui 1`] = `
<ListGroup
  flush={true}
  horizontal={false}
  tag="ul"
>
  <AttributeView
    label="Name"
  >
    42 BV
  </AttributeView>
  <AttributeView
    label="City"
  >
    Zoetermeer
  </AttributeView>
  <AttributeView
    label="Country"
  >
    Netherlands
  </AttributeView>
</ListGroup>
`;
